plugin/trino-ranger/src/main/java/io/trino/plugin/ranger/RangerTrinoEventListener.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.ranger;
+
+import io.trino.spi.eventlistener.EventListener;
+import io.trino.spi.eventlistener.QueryCompletedEvent;
+import io.trino.spi.eventlistener.QueryContext;
+import io.trino.spi.eventlistener.QueryCreatedEvent;
+import io.trino.spi.eventlistener.QueryMetadata;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+class RangerTrinoEventListener
+        implements EventListener
+{
+    private final Map<String, QueryCreatedEvent> activeQueries = new HashMap<>();
```

Comment:
It will be accessed from multiple threads, so doesn't that need to be a `ConcurrentMap`? (Or synchronized in all methods that use it.)